### PR TITLE
chore: bump ScreensExample to RN 74

### PR DESCRIPTION
**Pull request by alduzy**
## Description

PR bumping ScreensExample to RN 0.74.1

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Updated documentation: 
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/guides/GUIDE_FOR_LIBRARY_AUTHORS.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/native-stack/README.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/native-stack/types.tsx
- [ ] Ensured that CI passes

### CODE REVIEW

File: Example/android/app/src/main/java/com/swmansion/rnscreens/example/MainActivity.java [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-  protected ReactActivityDelegate createReactActivityDelegate() {
-    return new DefaultReactActivityDelegate(
-        this,
-        getMainComponentName(),
-        // If you opted-in for the New Architecture, we enable the Fabric Renderer.
-        DefaultNewArchitectureEntryPoint.getFabricEnabled());
-  }
-
-  @Override
-  protected void onCreate(Bundle savedInstanceState) {
```
Comment (WoLewicki) [inline Example/android/app/src/main/java/com/swmansion/rnscreens/example/MainActivity.java:35]: I think you forgot to transfer this method

File: Example/android/build.gradle [hunk 2]
```diff
@@ -32,3 +30,5 @@ allprojects {
         }
     }
 }
+
+apply plugin: "com.facebook.react.rootproject"
```
Comment (WoLewicki) [inline Example/android/build.gradle:34]: No newline

File: Example/App.tsx [hunk 3]
```diff
@@ -41,7 +41,7 @@ const SCREENS: Record<
   string,
   {
     title: string;
-    component: () => JSX.Element;
+    component: () => React.JSX.Element;
```
Comment (WoLewicki) [inline Example/App.tsx:44]: Why do we need `React` prefix here now?

File: Example/tsconfig.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "../tsconfig.json",
+  "extends": "@react-native/typescript-config/tsconfig.json",
```
Comment (WoLewicki) [inline Example/tsconfig.json:2]: Do you think we don't want to get the config still from the one above?

**Review by WoLewicki — COMMENTED** [review]
I added some comments. Looks good overall :tada:

File: Example/App.tsx (hunk 3 above)
Comment (alduzy) [reply Example/App.tsx:44]: That's what the upgrade helper suggested: https://react-native-community.github.io/upgrade-helper/?from=0.72.4&to=0.74.1#RnDiffApp-App.tsx

File: Example/android/build.gradle (hunk 2 above)
Comment (alduzy) [reply Example/android/build.gradle]: https://github.com/software-mansion/react-native-screens/pull/2137/commits/510594ae3d1a587b01757ad8c1ef3360e1977d7b

**Review by WoLewicki — APPROVED** [review]

**Pull request merged**